diff --git a/src/bliss_C.cc b/src/bliss_C.cc
index c000a66..8161381 100644
--- a/src/bliss_C.cc
+++ b/src/bliss_C.cc
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <assert.h>
 #include "graph.hh"
+#include "digraph.hh"
 extern "C" {
 #include "bliss_C.h"
 }
@@ -29,7 +30,7 @@ extern "C" {
  * \brief The true bliss graph is hiding in this struct.
  */
 struct bliss_graph_struct {
-  bliss::Graph* g;
+  bliss::AbstractGraph* g;
 };
 
 extern "C"
@@ -42,6 +43,16 @@ BlissGraph *bliss_new(const unsigned int n)
   return graph;
 }
 
+extern "C"
+BlissGraph *bliss_new_digraph(const unsigned int n)
+{
+  BlissGraph *graph = new bliss_graph_struct;
+  assert(graph);
+  graph->g = new bliss::Digraph(n);
+  assert(graph->g);
+  return graph;
+}
+
 extern "C"
 BlissGraph *bliss_read_dimacs(FILE *fp)
 {
@@ -103,16 +114,6 @@ void bliss_add_edge(BlissGraph *graph, unsigned int v1, unsigned int v2)
   graph->g->add_edge(v1, v2);
 }
 
-extern "C"
-int bliss_cmp(BlissGraph *graph1, BlissGraph *graph2)
-{
-  assert(graph1);
-  assert(graph1->g);
-  assert(graph2);
-  assert(graph2->g);
-  return (*graph1->g).cmp(*graph2->g);
-}
-
 extern "C"
 unsigned int bliss_hash(BlissGraph *graph)
 {
diff --git a/src/bliss_C.h b/src/bliss_C.h
index b144c05..ff914e6 100644
--- a/src/bliss_C.h
+++ b/src/bliss_C.h
@@ -71,6 +71,7 @@ typedef struct bliss_stats_struct
  * to add new vertices on-the-fly.
  */
 BlissGraph *bliss_new(const unsigned int N);
+BlissGraph *bliss_new_digraph(const unsigned int N);
 
 
 /**
@@ -132,17 +133,6 @@ unsigned int bliss_add_vertex(BlissGraph *graph, unsigned int c);
 void bliss_add_edge(BlissGraph *graph, unsigned int v1, unsigned int v2);
 
 
-/**
- * Compare two graphs according to a total order.
- * Return -1, 0, or 1 if the first graph was smaller than, equal to,
- * or greater than, resp., the other graph.
- * If 0 is returned, then the graphs have the same number vertices,
- * the vertices in them are colored in the same way, and they contain
- * the same edges; that is, the graphs are equal.
- */
-int bliss_cmp(BlissGraph *graph1, BlissGraph *graph2);
-
-
 /**
  * Get a hash value for the graph.
  */
